<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="19" passed="15" failed="4" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-08-12T21:15:31 IST" name="Suite" finished-at="2025-08-12T21:15:46 IST" duration-ms="14284">
    <groups>
      <group name="cucumber">
        <method signature="AbstractTestNGCucumberTests.runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" name="runScenario" class="io.cucumber.testng.AbstractTestNGCucumberTests"/>
      </group> <!-- cucumber -->
    </groups>
    <test started-at="2025-08-12T21:15:31 IST" name="Test" finished-at="2025-08-12T21:15:46 IST" duration-ms="14284">
      <class name="testcases.GetTrainTest">
        <test-method signature="getByTrainClass(java.lang.String)[pri:0, instance:testcases.GetTrainTest@359f7cdf]" started-at="2025-08-12T21:15:43 IST" name="getByTrainClass" description="Get booking data by trainClass" finished-at="2025-08-12T21:15:43 IST" duration-ms="268" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[THIRDAC]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getByTrainClass -->
      </class> <!-- testcases.GetTrainTest -->
      <class name="testcases.PostBookingTest">
        <test-method signature="getBooking()[pri:0, instance:testcases.PostBookingTest@1fa268de]" started-at="2025-08-12T21:15:43 IST" name="getBooking" description="Get response after adding bookings data" finished-at="2025-08-12T21:15:44 IST" duration-ms="280" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getBooking -->
        <test-method signature="postBooking()[pri:0, instance:testcases.PostBookingTest@1fa268de]" started-at="2025-08-12T21:15:44 IST" name="postBooking" description="Post booking data from BookingData.json" finished-at="2025-08-12T21:15:44 IST" duration-ms="356" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- postBooking -->
      </class> <!-- testcases.PostBookingTest -->
      <class name="testcases.DelInvalidIdTest">
        <test-method signature="deleteByInvalidId(int)[pri:0, instance:testcases.DelInvalidIdTest@4466af20]" started-at="2025-08-12T21:15:44 IST" name="deleteByInvalidId" description="Delete booking data by invalid bookingId" finished-at="2025-08-12T21:15:45 IST" duration-ms="258" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[909090]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response must be null!!! expected [null] but found [io.restassured.internal.RestAssuredResponseImpl@41f40dfa]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response must be null!!! expected [null] but found [io.restassured.internal.RestAssuredResponseImpl@41f40dfa]
at org.testng.Assert.fail(Assert.java:110)
at org.testng.Assert.failNotSame(Assert.java:1409)
at org.testng.Assert.assertNull(Assert.java:1342)
at testcases.DelInvalidIdTest.deleteByInvalidId(DelInvalidIdTest.java:20)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
at org.testng.SuiteRunner.run(SuiteRunner.java:330)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
at org.testng.TestNG.runSuites(TestNG.java:1099)
at org.testng.TestNG.run(TestNG.java:1067)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:308)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:71)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:113)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteByInvalidId -->
        <test-method signature="getBooking()[pri:0, instance:testcases.DelInvalidIdTest@4466af20]" started-at="2025-08-12T21:15:45 IST" name="getBooking" description="Get bookings data after deletion with invalid booking id" finished-at="2025-08-12T21:15:45 IST" duration-ms="294" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getBooking -->
      </class> <!-- testcases.DelInvalidIdTest -->
      <class name="testcases.DeleteIdTest">
        <test-method signature="deleteById(int)[pri:0, instance:testcases.DeleteIdTest@4f6ee6e4]" started-at="2025-08-12T21:15:44 IST" name="deleteById" description="Delete booking data by valid bookingId" finished-at="2025-08-12T21:15:44 IST" duration-ms="235" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1001]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteById -->
        <test-method signature="getBooking()[pri:0, instance:testcases.DeleteIdTest@4f6ee6e4]" started-at="2025-08-12T21:15:44 IST" name="getBooking" description="Get bookings data after deletion with valid booking id" finished-at="2025-08-12T21:15:44 IST" duration-ms="278" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getBooking -->
      </class> <!-- testcases.DeleteIdTest -->
      <class name="testcases.GetIdTest">
        <test-method signature="getById(int)[pri:0, instance:testcases.GetIdTest@a514af7]" started-at="2025-08-12T21:15:45 IST" name="getById" description="Get booking data by bookingId" finished-at="2025-08-12T21:15:45 IST" duration-ms="289" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1001]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getById -->
      </class> <!-- testcases.GetIdTest -->
      <class name="testcases.AuthAccessTest">
        <test-method signature="getAccessToken()[pri:0, instance:testcases.AuthAccessTest@6356695f]" started-at="2025-08-12T21:15:43 IST" name="getAccessToken" description="To get access_token with valid credentials" finished-at="2025-08-12T21:15:43 IST" duration-ms="14" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getAccessToken -->
      </class> <!-- testcases.AuthAccessTest -->
      <class name="testcases.GetListTest">
        <test-method signature="getList()[pri:0, instance:testcases.GetListTest@4f18837a]" started-at="2025-08-12T21:15:43 IST" name="getList" description="Get the bookings data from train API" finished-at="2025-08-12T21:15:43 IST" duration-ms="281" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getList -->
      </class> <!-- testcases.GetListTest -->
      <class name="testrunner.CucumberTestNGRunner">
        <test-method is-config="true" signature="setUpClass(org.testng.ITestContext)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:32 IST" name="setUpClass" finished-at="2025-08-12T21:15:33 IST" duration-ms="883" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[org.testng.TestRunner@3093b9ff]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUpClass -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:33 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:39 IST" duration-ms="6411" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate add booking from JSON - post request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:39 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:40 IST" duration-ms="473" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate view booking list - get request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:40 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:40 IST" duration-ms="359" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate view booking by valid Id - get request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:40 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:41 IST" duration-ms="388" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate view booking by valid trainClass - get request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:41 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:41 IST" duration-ms="281" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate view booking by invalid Id - get request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response is not JSON!!!
Expected: a string containing "application/json"
     but: was ""]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response is not JSON!!!
Expected: a string containing "application/json"
     but: was ""
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at utils.ValidationUtils.validateResponse(ValidationUtils.java:15)
at stepdefinations.GetInvalidSteps.validate_the_get_by_invalidid_response(GetInvalidSteps.java:26)
at ✽.validate the get by invalidid response(file:///C:/Users/DELL/eclipse-workspace/RestAssuredTestingCucumber/src/test/resources/feature/BookingAPI.feature:30)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:41 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:41 IST" duration-ms="337" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate delete booking by valid Id - delete request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:41 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:42 IST" duration-ms="379" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate delete booking by invalid Id - delete request"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Booking Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[StatusCode mismatch!!!
Expected: <404>
     but: was <200>]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: StatusCode mismatch!!!
Expected: <404>
     but: was <200>
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at utils.ValidationUtils.validateStatusCode(ValidationUtils.java:21)
at stepdefinations.DelInvalidIdSteps.validate_the_del_by_invalidid_response(DelInvalidIdSteps.java:29)
at ✽.validate the del by invalidid response(file:///C:/Users/DELL/eclipse-workspace/RestAssuredTestingCucumber/src/test/resources/feature/BookingAPI.feature:40)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:42 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-08-12T21:15:43 IST" duration-ms="888" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["To validate auth code and access token"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Authorization Functionality RestAssured API"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method is-config="true" signature="tearDownClass()[pri:0, instance:testrunner.CucumberTestNGRunner@3bb9a3ff]" started-at="2025-08-12T21:15:43 IST" name="tearDownClass" finished-at="2025-08-12T21:15:43 IST" duration-ms="136" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDownClass -->
      </class> <!-- testrunner.CucumberTestNGRunner -->
      <class name="testcases.GetInvalidIdTest">
        <test-method signature="getByInvalidId(int)[pri:0, instance:testcases.GetInvalidIdTest@6b927fb]" started-at="2025-08-12T21:15:45 IST" name="getByInvalidId" description="Get booking data by invalid bookingId" finished-at="2025-08-12T21:15:46 IST" duration-ms="296" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[909090]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response must be null!!! expected [null] but found [io.restassured.internal.RestAssuredResponseImpl@6fcc0a1]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response must be null!!! expected [null] but found [io.restassured.internal.RestAssuredResponseImpl@6fcc0a1]
at org.testng.Assert.fail(Assert.java:110)
at org.testng.Assert.failNotSame(Assert.java:1409)
at org.testng.Assert.assertNull(Assert.java:1342)
at testcases.GetInvalidIdTest.getByInvalidId(GetInvalidIdTest.java:20)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
at org.testng.SuiteRunner.run(SuiteRunner.java:330)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
at org.testng.TestNG.runSuites(TestNG.java:1099)
at org.testng.TestNG.run(TestNG.java:1067)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:308)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:71)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:113)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getByInvalidId -->
      </class> <!-- testcases.GetInvalidIdTest -->
    </test> <!-- Test -->
  </suite> <!-- Suite -->
</testng-results>
